roofit/roofit/src/RooMultiPdf.cxx

```diff
@@ -0,0 +1,95 @@
+ *   RW, Ruddick William  UC Colorado        wor@slac.stanford.edu           *
+ *                                                                           *
+ * Copyright (c) 2000-2005, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+/** \class RooMultiPdf
+   \ingroup Roofit
+
+   The class RooMultiPdf allows for the creation of a RooMultiPdf object,
+   which can switch between previously set probability density functions (PDF)
+   by setting their associated indices.*/
+
+#include "../inc/RooMultiPdf.h"
+
+#include "RooConstVar.h"
+
+ClassImp(RooMultiPdf)
+   // Constructing a RooMultiPdf
+   //  parameter name : The name of the RooMultiPdf object
+   //  parameter title : Display title in plots
+   //  parameter _x :  variable used to select which PDF that selects the active PDF.
+   //  parameter _c : A list of the pdfs.The index of each PDF in the list should match the values in _x
+   //_____________________________________________________________________________
+   RooMultiPdf::RooMultiPdf(const char *name, const char *title, RooCategory &_x, const RooArgList &_c)
+   : RooAbsPdf(name, title), // call of constructor base class RooAbsPdf passing it name and title
+     c("_pdfs", "The list of pdfs", this),
+     corr("_corrs", "The list of correction factors", this),
+     x("_index", "the pdf index", this, _x)
+
+// parameter corr : Holds correction factors - number of free parameters
+// in each PDF held in the RooMultiPdf object
+{
+   int count = 0;
+
+   c.add(_c);
+   for (RooAbsArg *pdf : c) {
+      // This is done by the user BUT is there a way to do it at construction?
+      _x.defineType(
+         ("_pdf" + std::to_string(count)).c_str(),
+         count); // For each PDF, defines a new type in the index category _x, associating "pdf0", "pdf1", etc. to count
+      std::unique_ptr<RooArgSet> variables(
+         pdf->getVariables()); // getVariables() gets all the variables that the PDF depends on
+      std::unique_ptr<RooAbsCollection> nonConstVariables(variables->selectByAttrib(
+         "Constant", false)); // selectByAttrib("Constant", false) filters out parameters that are not constant
+      // Isn't there a better way to hold on to these values?
+      RooConstVar *tmp = new RooConstVar((std::string{"const"} + pdf->GetName()).c_str(), "",
+                                         nonConstVariables->size()); // records how many  parameters the PDF has
+      corr.addOwned(*tmp);
```

Comment:
Please prefer the interface with `std::unique_ptr` (the raw pointer interfaces will be eventually deprecated).
Also, I would suggest to put the comments in a separate line, so there are no awkward line breaks by the code formatter.
```suggestion
      // records how many  parameters the PDF has
      corr.addOwned(
         std::make_unique<RooConstVar>((std::string{"const"} + pdf->GetName()).c_str(), "", nonConstVariables->size()));
```